src/v/storage/segment_appender.cc

```diff
@@ -87,8 +87,7 @@ segment_appender::segment_appender(segment_appender&& o) noexcept
   , _prev_head_write(std::move(o._prev_head_write))
   , _inflight(std::move(o._inflight))
   , _callbacks(std::exchange(o._callbacks, nullptr))
-  , _inactive_timer([this] { handle_inactive_timer(); })
-  , _previously_inactive(o._previously_inactive) {
+  , _inactive_timer([this] { handle_inactive_timer(); }) {
```

Comment:
what's interesting is now this has a downside of during idle times, a bunch of the idle partitions cannot leverage free memory. in effect every partition locks 32KB of memory. 

wouldn't you want to preserve this? I mean it's OK for now, but seems like we lost some functionality. 

I'm ok w/ removing this, but the commit should say that explicitly. This *removes* functionality of giving back memory to the shared pool.

Reply:
I'm not following along with what functionality you're saying is being removed.